### PR TITLE
[Development] Remove 526 unneeded reserve data

### PR DESCRIPTION
**Pull request by Mottie**
## Description

At times, some users are unable to submit form 21-526EZ  due to a missing reserve `unitName`. It was discovered that the user had not entered any reserve information into the form, and that the prefill data `reservesNationalGuardService` was included (with a missing `unitName`) in the submission. It was found:
* In one case, the user added mock reserve data into the online form, then removed it, and then was able to submit their application 
* In another case, the user added their reserve information, answered the federal orders question, and then was able to submit their application

This PR, removes the extra `reservesNationalGuardService` when no reserve or national guard service period was entered into the form.

Related issues:
* Discovery: https://github.com/department-of-veterans-affairs/va.gov-team/issues/6797
* Implementation: https://github.com/department-of-veterans-affairs/va.gov-team/issues/7335

## Testing done

Added `service-info.json` data & transformed data for the submission transformer to test.

## Acceptance criteria
- [ ] `reservesNationalGuardService` data is removed when associated `serviceInformation` is not provided

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by ndsprinkle — APPROVED** [review]

File: src/applications/disability-benefits/all-claims/submit-transformer.js [hunk 1]
```diff
@@ -220,6 +220,19 @@ export function transform(formConfig, form) {
 
   const filterRatedViewFields = formData => filterViewFields(formData);
 
+  const filterServicePeriods = formData => {
+    const { serviceInformation } = formData;
+    if (!serviceInformation || hasGuardOrReservePeriod(serviceInformation)) {
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/submit-transformer.js]: Ooh, tricky. It _looks_ like [`hasGuardOrReservePeriod`](https://github.com/department-of-veterans-affairs/vets-website/blob/remove-webpack-from-metalsmith-cv/src/applications/disability-benefits/all-claims/utils.jsx#L84) takes the `formData`, not the `serviceInformation`, but it _lies_. Sneaky, sneaky.

File: src/applications/disability-benefits/all-claims/tests/data/service-info.json [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        "view:isTitle10Activated": false,
+        "obligationTermOfServiceDateRange": {
+          "from": "2007-05-22",
+          "to": "2008-06-05"
+        },
+        "unitName": "Unit name here"
+      },
+      "servicePeriods": [
+        {
+          "serviceBranch": "Air Force",
```
Comment (Mottie) [inline src/applications/disability-benefits/all-claims/tests/data/service-info.json]: Copied from the minimal test json, removed "Reserved" from this line

File: src/applications/disability-benefits/all-claims/tests/data/transformed-data/service-info.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      }
+    ],
+    "serviceInformation": {
+      "servicePeriods": [
+        {
+          "serviceBranch": "Air Force",
+          "dateRange": { "from": "2001-03-21", "to": "2014-07-21" }
+        }
+      ]
+    },
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/tests/data/transformed-data/service-info.json]: This is good, but I might recommend checking for the removal of the data in a unit test instead. Like this, it'll run the puppeteer form app tester and plugin all the data, which should work perfectly, but is relatively slow.

File: src/applications/disability-benefits/all-claims/utils.jsx [hunk 4]
```diff
@@ -90,7 +90,7 @@ export const hasGuardOrReservePeriod = formData => {
   return serviceHistory.reduce((isGuardReserve, { serviceBranch }) => {
     // For a new service period, service branch defaults to undefined
     if (!serviceBranch) {
-      return false;
+      return isGuardReserve;
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/utils.jsx]: Ooh, good call.

**Review by cvalarida — APPROVED** [review]
Looks good from where I sit! My only recommendation is to use a faster unit test instead of the slower puppeteer test to test the submit transformer. (Test test test. How many times do you think I can say "test" in one review?)

File: src/applications/disability-benefits/all-claims/tests/data/transformed-data/service-info.json (hunk 3 above)
Comment (Mottie) [reply src/applications/disability-benefits/all-claims/tests/data/transformed-data/service-info.json]: Oh, I didn't know these were also run on e2e tests. I've just updated this PR.

Comment (cvalarida) [reply src/applications/disability-benefits/all-claims/tests/data/transformed-data/service-info.json]: Also? How did you expect them to be run? I might be misunderstanding something.

Comment (Mottie) [reply src/applications/disability-benefits/all-claims/tests/data/transformed-data/service-info.json]: The submit transformer unit test loads in each json file & transformed json file, then does a comparison - see https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/disability-benefits/all-claims/tests/submit-transformer.unit.spec.js#L30-L56

**Pull request merged**